Source/WebCore/editing/Editing.cpp

```diff
@@ -1222,4 +1224,130 @@ HashSet<RefPtr<HTMLImageElement>> visibleImageElementsInRangeWithNonLoadedImages
     return result;
 }
 
+static std::optional<unsigned> visualDistanceOnSameLine(const RenderedPosition& first, const RenderedPosition& second)
+{
+    if (first.isNull() || second.isNull())
+        return std::nullopt;
+
+    if (first.box() == second.box())
+        return std::max(first.offset(), second.offset()) - std::min(first.offset(), second.offset());
+
+    auto distanceToBoundary = [](const RenderedPosition& position, bool leftSide) {
```

Comment:
"leftSide" isn't always left side, is it?

Reply:
Ah, so for this function, the `leftSide` parameter represents whether to count from the visual left boundary of the text run, or the visual right boundary. The `leftSide == (box->direction() == TextDirection::LTR)` check below then accounts for the difference in text direction (e.g., "the distance from the left boundary to the position offset in RTL text is computed by subtracting the (logical) max offset from the position offset").

To help clarify this, I'll refactor the code:
- Rename `distanceToBoundary` → `distanceFromOffsetToVisualBoundary`
- Use `enum class VisualBoundary` with values `Left/Right` instead of a `bool`.

Thanks for the review!